modules/020-deckhouse/docs/internal/RELEASE.md

```diff
@@ -0,0 +1,23 @@
+# DeckhouseRelease details
+
+## Available annotations
+
+| annotation                               | description                                                                        |
+|------------------------------------------|------------------------------------------------------------------------------------|
+| release.deckhouse.io/force               | Apply specified release without any checks. Force deploy.                          |
+| release.deckhouse.io/disruption-approved | Approve release with disruptive changes. Works if update.disruptionMode is Manual. |
+| release.deckhouse.io/approved            | Approve release for deployment. Works if update.mode is Manual.                    |
+
+## Disruptive release
+
+It's a release with some potentially dangerous changes (change some default value / behavior / docker -> containerd, etc)
+To handle this release, you should add disruption check logic in a release `X-1`, for example - register requirements.DisruptionFunc in init() [example](modules/402-ingress-nginx/hooks/requirements.go)
+And then in a release `X` you should add record `"disruption:$functionName": "true"` to the [requirements.json](requirements.json) file.
+For example:
+In the release 1.35.0 set disruption check logic into ingressNginx
+In the release 1.36.0 set `"disruption:ingressNginx": "true"` to the requirements.json
```

Comment:
Can I add both check and disruption to 1.36.0? What's the point in splitting check and disruption between different versions?

Reply:
You could have
```
"ingressNginx": "1.1",
"disruption:ingressNginx": "true"
```
in the same release. It would work, cause these checks pass in a row

In my mind they are splitted like that:
 - RequirementCheck (like `"ingressNginx": "1.1"`, or `"k8s": "1.19"`) - it's some strict check of outer dependencies which deny a release deploy until a requirement is met. (**hard block**)
 - DisruptionCheck - it's a check, that warn user about some meaningful changes, which are controlled by our code and logic (simple: we are changing some behavior) (**soft block**)

Reply:
So, in order to have Deckhouse in working state, we need to precaution clients in `X-1` release, because if the requirement is not met, release `X` will not work. The client will be notified, but Deckhouse will not be stuck because of the inevitable migration. I see.